### PR TITLE
Add Compiling.md

### PR DESCRIPTION
**Pull request by ManpreetSingh2004**

### CODE REVIEW

File: COMPILING.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+- Install the required dependencies using the following command:
+
+    ```bash
+    pacman -S mingw-w64-cross-binutils mingw-w64-x86_64-toolchain mingw-w64-x86_64-scons mingw-w64-x86_64-yasm diffutils make
+    ```
+
+## Step 2: Cloning repositories
+
+- Open a terminal in the desired directory and clone the Gozen repository
```
Comment (aappaapp) [inline COMPILING.md:30]: It should be GoZen (with the capital letter Z), not Gozen.

Comment (TheRealSlander) [reply COMPILING.md:30]: Nice catch!

Comment (voylin) [inline COMPILING.md:30]:
```suggestion
- Open a terminal in the desired directory and clone the GoZen repository
```

File: COMPILING.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Compiling GoZen
+
+This guide will help you compile the GoZen gdextension.
+
+## Step 1: Installing dependencies
+
+### Linux
+
```
Comment (voylin) [inline COMPILING.md:8]:
```suggestion

- For dependencies you will mainly need Python and scons. These are needed for compiling the GDExtension which gives access to FFmpeg, which also needs to be installed.

    ```bash
    pacman -Syu python scons ffmpeg
    ```
```

**Review by voylin — CHANGES_REQUESTED** [review]
Addition for the dependencies on Linux.

File: COMPILING.md (hunk 1 above)
Comment (voylin) [reply COMPILING.md:30]: Added this to the review in which I also added the info for dependencies for Linux.

File: COMPILING.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+- For dependencies you will mainly need Python and scons. These are needed for compiling the GDExtension which gives access to FFmpeg, which also needs to be installed.
```
Comment (aappaapp) [inline COMPILING.md:9]:
```suggestion
- For dependencies you will mainly need Python and SCons. These are needed for compiling the GDExtension which gives access to FFmpeg, which also needs to be installed.
```

File: COMPILING.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    ```bash
+    pacman -Syu python scons ffmpeg
+    ```
+### Windows
+
+- Install Msys2 from [www.msys2.org](https://www.msys2.org/)
```
Comment (aappaapp) [inline COMPILING.md:16]:
```suggestion
- Install MSYS2 from [www.msys2.org](https://www.msys2.org/)
```

File: COMPILING.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+## Step 2: Cloning repositories
+
+- Open a terminal in the desired directory and clone the GoZen repository
+
+    ```bash
+    git clone --recurse-submodules https://github.com/VoylinsGamedevJourney/GoZen.git
+    ```
+
+- Move to the Gozen directory
```
Comment (aappaapp) [inline COMPILING.md:41]:
```suggestion
- Move to the GoZen directory
```

File: COMPILING.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+    ```bash
+    cd GoZen
+    ```
+
+- Clone the Godot-cpp repository
```
Comment (aappaapp) [inline COMPILING.md:47]:
```suggestion
- Clone the godot-cpp repository
```

File: COMPILING.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+## Step 3: Compiling and editing the project
+
+- Execute the build script
+
+    ```bash
+    sh build.sh
+    ```
+
+    **Note:** You will need to compile ffmpeg at least once. Enter `y` when prompted.
```
Comment (aappaapp) [inline COMPILING.md:61]:
```suggestion
    **Note:** You will need to compile FFmpeg at least once. Enter `y` when prompted.
```

**Review by aappaapp — CHANGES_REQUESTED** [review]

**Review by aappaapp — APPROVED** [review]
Looks good to me!

**Comment by voylin** [comment]
Thank you!

**Comment by ManpreetSingh2004** [comment]
Thank you all so much for your incredible support and collaboration in getting this pull request merged!

**Pull request merged**